[docs] "What's new in Svelte" July newsletter (#7637)
sveltejs/svelte: site/content/blog/2022-07-01-whats-new-in-svelte-july-2022.md

---
title: "What's new in Svelte: July 2022"
description: "Faster SSR, Language tools improvements and something special..."
author: Daniel Sandoval
authorURL: https://desandoval.net
---

> With [something special](https://twitter.com/SvelteSociety/status/1539284972544180230?s=20&t=3Qoi371NrL9LP6xMOF63ag) dropping today along with faster SSR and a number of new language tools features, there's a lot to cover in this month's newsletter...

Reviewer: "a number of new language tools features" is kind of vague - maybe swap it for something a bit more specific?
```suggestion
With [something special](https://twitter.com/SvelteSociety/status/1539284972544180230?s=20&t=3Qoi371NrL9LP6xMOF63ag) dropping today along with faster SSR and support for Vitest and Storybook in SvelteKit, there's a lot to cover in this month's newsletter...
```

So let's dive in!

> ## What's new in Svelte & Language Tools

Reviewer: I thought it might be useful to share an update about OpenCollective funding

```suggestion
## OpenCollective funding drives Svelte forward

Svelte supporters have donated approximately $80,000 to [the project on OpenCollective](https://opencollective.com/svelte). We're happy to share that the funds are being drawn on to move Svelte forward in a meaningful way. [@gtm-nayan](https://github.com/gtm-nayan) has begun triaging and fixing SvelteKit issues this past month as a paid contributor to the project to help us get SvelteKit to a 1.0 level of stability! @gtm-nayan has been an active member of the Svelte community for quite some time and is well known for writing the bot that helps keep our Discord server running. We're happy that this funding has allowed Svelte to get much more of his time.

We will also be utilizing OpenCollective funds to allow Svelte core maintainers to attend Svelte Summit in person this fall.

## What's new in Svelte & Language Tools
```

> - [learn.svelte.dev](https://learn.svelte.dev/) is a new way to learn Svelte and SvelteKit from the ground up!

Reviewer: ```suggestion
- [learn.svelte.dev](https://learn.svelte.dev/) is a new way to learn Svelte and SvelteKit from the ground up that is currently in development!
```

- Faster SSR is coming in the next Svelte release. A PR two years in the making, resulting in up to 3.05x faster in some tests! ([PR](https://github.com/sveltejs/svelte/pull/5701))
- "Find File References" in the latest version of the Svelte extension shows where Svelte files have been imported ([0.14.29](https://github.com/sveltejs/language-tools/releases/tag/language-server-0.14.29), [Demo](https://twitter.com/dummdidumm_/status/1532459709604716544/photo/1))
- The Svelte extension now supports CSS path completion ([0.14.29](https://github.com/sveltejs/language-tools/releases/tag/language-server-0.14.29))

## What's new in SvelteKit

> - Static assets can now be symlinked in development environments ([#5089](https://github.com/sveltejs/kit/pull/5089))

Reviewer: ```suggestion
- The big change in SvelteKit this month is the introduction of `@sveltejs/kit/experimental/vite` which allows SvelteKit to interoperate with other tools in the Vite ecosystem like Vitest and Storybook ([#5094](https://github.com/sveltejs/kit/pull/5094)). Please [leave feedback](https://github.com/sveltejs/kit/issues/5184) as to whether the feature works and is helpful as we consider taking it out of experimental and making `vite.config.js` required for all users
- Static assets can now be symlinked in development environments ([#5089](https://github.com/sveltejs/kit/pull/5089))
```

- [Pronauns](https://www.pronauns.com) helps you learn pronunciation online with IPA to speak better and sound more native
- [Immich](https://www.immich.app/) is an open source, high performance self-hosted backup solution for videos and photos on your mobile phone
- [Pendek](https://github.com/leovoon/link-shortener) is a link shortener built with SvelteKit, Prisma and PlanetScale
- [Grunfy](https://grunfy.com/tools) is a set of guitar tools - recently migrated to SvelteKit
- [Radiant: The Future of Radio](https://play.google.com/store/apps/details?id=co.broadcastapp.Radiant) is a personal radio station app built with Svelte and Capacitor
- [Imperfect Reminders](https://imperfectreminders.mildlyupset.com/) is a todo list for things that are only sort of time sensitive
- [Svelvet](https://github.com/open-source-labs/Svelvet) is a lightweight Svelte component library for building interactive node-based diagrams
- [publint](https://github.com/bluwy/publint) lints for packaging errors to ensure compatibility across environments
- [Playlistr](https://github.com/alextana/spotify-playlist-creator) helps manage and create Spotify playlists

> - [Geoff Rich's page transitions demo](https://twitter.com/geoffrich_/status/1534980702785003520) shows how SvelteKit's beforeNavigate/afterNavigate hooks can make smooth document transitions in the latest Chrome Canary

Reviewer: ```suggestion
- [Geoff Rich's page transitions demo](https://twitter.com/geoffrich_/status/1534980702785003520) shows how SvelteKit's `beforeNavigate`/`afterNavigate` hooks can make smooth document transitions in the latest Chrome Canary
```

_To Read_
- [Up-To-Date Analytics on a Static Website](https://paullj.github.io/posts/up-to-date-analytics-on-a-static-website) and [Fast, Lightweight Fuzzy Search using Fuse.js](https://paullj.github.io/posts/fast-lightweight-fuzzy-search-using-fuse.js) by paullj
- [Use SvelteKit as a handler in the ExpressJs project](https://chientrm.medium.com/use-sveltekit-as-a-handler-in-the-expressjs-project-15524b01128f) by Tran Chien
- [Creating a desktop application with Tauri and SvelteKit](https://github.com/Stijn-B/tauri-sveltekit-example) by Stijn-B
- [List of awesome Svelte stores](https://github.com/samuba/awesome-svelte-stores) by samuba
- [SvelteKit Content Security Policy: CSP for XSS Protection](https://rodneylab.com/sveltekit-content-security-policy/) by Rodney Lab
- [SvelteKit Hooks. Everything You Need To Know](https://kudadam.com/blog/understanding-sveltekit-hooks) by Lucretius K. Biah
- [3 tips for upgrading the performance of your Svelte stores](https://www.mathiaspicker.com/posts/3-tips-for-upgrading-the-performance-of-your-svelte-stores) by Mathias Picker

> _To Hear_

Reviewer 2: If there is nothing in _To Hear_ this should be removed

---
title: "What's new in Svelte: July 2022"
description: "Faster SSR, language tools improvements and something special..."
author: Daniel Sandoval
authorURL: https://desandoval.net
---

> With [something special](https://twitter.com/SvelteSociety/status/1539284972544180230?s=20&t=3Qoi371NrL9LP6xMOF63ag) dropping today along with faster SSR and support for Vitest and Storybook in SvelteKit, there's a lot to cover in this month's newsletter...

Reviewer 3: I think I'd wait until it's actually dropped before releasing the new blog post 😄 otherwise great as always!

Author: I've got some copy in the wings waiting to plug in here ;) Will drop it in right before the merge on the first :)

Author: After a conversation with @kevmodrome, we're going to take out the "something special": https://github.com/sveltejs/svelte/pull/7637/commits/ad6114d28e3c0bb9cbecf4c920098cb99d1b0958

He needs more time to polish it up ;)

We will also be utilizing OpenCollective funds to allow Svelte core maintainers to attend [Svelte Summit](https://www.sveltesummit.com/) in person this fall. Thanks to everyone who has donated, so far!

## What's new in Svelte & Language Tools
- [learn.svelte.dev](https://learn.svelte.dev/) is a new way to learn Svelte and SvelteKit from the ground up that is currently in development
- Faster SSR is coming in the next Svelte release. A PR two years in the making, resulting in up to 3.05x faster in some tests! ([PR](https://github.com/sveltejs/svelte/pull/5701))

> - "Find File References" in the latest version of the Svelte extension shows where Svelte files have been imported ([0.14.29](https://github.com/sveltejs/language-tools/releases/tag/language-server-0.14.29), [Demo](https://twitter.com/dummdidumm_/status/1532459709604716544/photo/1))

Reviewer 3: We also released "find component references" recently, not sure if it's too similar to mention, or if it makes sense to mention it in this same bullet point

- The Svelte extension now supports CSS path completion ([0.14.29](https://github.com/sveltejs/language-tools/releases/tag/language-server-0.14.29))

## What's new in SvelteKit
- Introduced `@sveltejs/kit/experimental/vite` which allows SvelteKit to interoperate with other tools in the Vite ecosystem like Vitest and Storybook ([#5094](https://github.com/sveltejs/kit/pull/5094)). Please [leave feedback](https://github.com/sveltejs/kit/issues/5184) as to whether the feature works and is helpful as we consider taking it out of experimental and making `vite.config.js` required for all users

> - Switched to the Undici `fetch` implementation ([#5117](https://github.com/sveltejs/kit/pull/5117)), which unblocks implementing streaming

Reviewer: streaming has been merged now
```suggestion
- Steaming in endpoints is now supported ([#3419](https://github.com/sveltejs/kit/issues/3419)). This was enabled by switching to the Undici `fetch` implementation ([#5117](https://github.com/sveltejs/kit/pull/5117))
```

---
title: "What's new in Svelte: July 2022"
description: "Faster SSR, language tools improvements and something special..."
author: Daniel Sandoval
authorURL: https://desandoval.net
---

With [something special](https://twitter.com/SvelteSociety/status/1539284972544180230?s=20&t=3Qoi371NrL9LP6xMOF63ag) dropping today along with faster SSR and support for Vitest and Storybook in SvelteKit, there's a lot to cover in this month's newsletter...

So let's dive in!

## OpenCollective funding drives Svelte forward

Svelte supporters have donated approximately $80,000 to [the project on OpenCollective](https://opencollective.com/svelte). We're happy to share that the funds are being drawn on to move Svelte forward in a meaningful way. [@gtm-nayan](https://github.com/gtm-nayan) has begun triaging and fixing SvelteKit issues this past month as a paid contributor to the project to help us get SvelteKit to a 1.0 level of stability! **@gtm-nayan** has been an active member of the Svelte community for quite some time and is well known for writing the bot that helps keep our Discord server running. We're happy that this funding has allowed Svelte to get much more of his time.

> We will also be utilizing OpenCollective funds to allow Svelte core maintainers to attend [Svelte Summit](https://www.sveltesummit.com/) in person this fall. Thanks to everyone who has donated, so far!

Reviewer: ```suggestion
We will also be utilizing OpenCollective funds to allow Svelte core maintainers to attend [Svelte Summit](https://www.sveltesummit.com/) in person this fall. Thanks to everyone who has donated so far!
```

## What's new in Svelte & Language Tools
- [learn.svelte.dev](https://learn.svelte.dev/) is a new way to learn Svelte and SvelteKit from the ground up that is currently in development

> - Faster SSR is coming in the next Svelte release. A PR two years in the making, resulting in up to 3.05x faster in some tests! ([PR](https://github.com/sveltejs/svelte/pull/5701))

Reviewer: ```suggestion
- Faster SSR is coming in the next Svelte release. A PR two years in the making, resulting in up to 3x faster rendering in some benchmarking tests! ([PR](https://github.com/sveltejs/svelte/pull/5701))
```

- "Find File References" ([0.14.28](https://github.com/sveltejs/language-tools/releases/tag/language-server-0.14.28)) and "Find Component References" ([0.14.29](https://github.com/sveltejs/language-tools/releases/tag/language-server-0.14.29)) in the latest versions of the Svelte extension shows where Svelte files and components have been imported and used ([Demo](https://twitter.com/dummdidumm_/status/1532459709604716544/photo/1))
- The Svelte extension now supports CSS path completion ([0.14.29](https://github.com/sveltejs/language-tools/releases/tag/language-server-0.14.29))

## What's new in SvelteKit

> - Introduced `@sveltejs/kit/experimental/vite` which allows SvelteKit to interoperate with other tools in the Vite ecosystem like Vitest and Storybook ([#5094](https://github.com/sveltejs/kit/pull/5094)). Please [leave feedback](https://github.com/sveltejs/kit/issues/5184) as to whether the feature works and is helpful as we consider taking it out of experimental and making `vite.config.js` required for all users

Reviewer: I'm hoping to get https://github.com/sveltejs/kit/pull/5332 merged before this goes out so that we can remove the `experimental` portion :smile:

Author: Still open at this time, so let's merge without it :)

- [Pendek](https://github.com/leovoon/link-shortener) is a link shortener built with SvelteKit, Prisma and PlanetScale
- [Grunfy](https://grunfy.com/tools) is a set of guitar tools - recently migrated to SvelteKit
- [Radiant: The Future of Radio](https://play.google.com/store/apps/details?id=co.broadcastapp.Radiant) is a personal radio station app built with Svelte and Capacitor
- [Imperfect Reminders](https://imperfectreminders.mildlyupset.com/) is a todo list for things that are only sort of time sensitive
- [Svelvet](https://github.com/open-source-labs/Svelvet) is a lightweight Svelte component library for building interactive node-based diagrams
- [publint](https://github.com/bluwy/publint) lints for packaging errors to ensure compatibility across environments
- [Playlistr](https://github.com/alextana/spotify-playlist-creator) helps manage and create Spotify playlists
- [Geoff Rich's page transitions demo](https://twitter.com/geoffrich_/status/1534980702785003520) shows how SvelteKit's `beforeNavigate`/`afterNavigate` hooks can make smooth document transitions in the latest Chrome Canary

> - [Menger Sponge](https://twitter.com/a_warnes/status/1536215896078811137) is a fractal built with Threlte

Reviewer 4: I'd like to pitch https://ptable-elements.netlify.app for the community show case section.

Want to contribute to a site using the latest SvelteKit features? [Help build the Svelte Society site](https://github.com/svelte-society/sveltesociety.dev/issues)!

**Learning Resources**

> _Featuring Svelte Society_

Reviewer: I wonder if there's a better heading for this. E.g. the Full Stack Documentation talk isn't really featuring Svelte Society. Maybe something like "Created by the Svelte team" (the documentary is primarily created by OfferZen, but the Svelte Team certainly played a role in its creation)

Reviewer 3: "Starring the Svelte team"?

---
title: "What's new in Svelte: July 2022"
description: "Faster SSR, language tools improvements and a new paid contributor!"
author: Daniel Sandoval
authorURL: https://desandoval.net
---

From faster SSR to support for Vitest and Storybook in SvelteKit, there's a lot to cover in this month's newsletter...

So let's dive in!

## OpenCollective funding drives Svelte forward

> Svelte supporters have donated approximately $80,000 to [the project on OpenCollective](https://opencollective.com/svelte). We're happy to share that the funds are being drawn on to move Svelte forward in a meaningful way. **[@gtm-nayan](https://github.com/gtm-nayan)** has begun triaging and fixing SvelteKit issues this past month as a paid contributor to the project to help us get SvelteKit to a 1.0 level of stability! @gtm-nayan has been an active member of the Svelte community for quite some time and is well known for writing the bot that helps keep our Discord server running. We're happy that this funding has allowed Svelte to get much more of his time.

Reviewer 5: `@gtm-nayan` not sure if it will render correctly in page if there is `@` without any "fences".

Author: LGTM:
<img width="929" alt="image" src="https://user-images.githubusercontent.com/7872348/176940984-18715f49-3df9-46e3-89e5-f99d798b1016.png">
